cmd/aries-js-worker/src/aries.js

```diff
@@ -184,71 +180,71 @@ export const Aries = function(opts) {
-                return invoke(this.pkgname, "RegisteredServices", text, "timeout while getting list of registered services")
+                return invoke(aw, pending,  this.pkgname, "RegisteredServices", text, "timeout while getting list of registered services")
             },
             registerMessageService: async function (text) {
-                return invoke(this.pkgname, "RegisterMessageService", text, "timeout while registering service")
+                return invoke(aw, pending,  this.pkgname, "RegisterMessageService", text, "timeout while registering service")
             },
             registerHTTPMessageService: async function (text) {
-                return invoke(this.pkgname, "RegisterHTTPMessageService", text, "timeout while registering HTTP service")
+                return invoke(aw, pending,  this.pkgname, "RegisterHTTPMessageService", text, "timeout while registering HTTP service")
             },
             unregisterMessageService: async function (text) {
-                return invoke(this.pkgname, "UnregisterMessageService", text, "timeout while unregistering service")
+                return invoke(aw, pending,  this.pkgname, "UnregisterMessageService", text, "timeout while unregistering service")
             },
             sendNewMessage: async function (text) {
-                return invoke(this.pkgname, "SendNewMessage", text, "timeout while sending new message")
+                return invoke(aw, pending,  this.pkgname, "SendNewMessage", text, "timeout while sending new message")
             },
             sendReplyMessage: async function (text) {
-                return invoke(this.pkgname, "SendReplyMessage", text, "timeout while sending reply message")
+                return invoke(aw, pending,  this.pkgname, "SendReplyMessage", text, "timeout while sending reply message")
             }
         },
 
         vdri: {
             pkgname: "vdri",
             createPublicDID: async function (text) {
-                return invoke(this.pkgname, "CreatePublicDID", text, "timeout while creating public DID")
+                return invoke(aw, pending,  this.pkgname, "CreatePublicDID", text, "timeout while creating public DID")
             },
         },
 
         router: {
             pkgname: "router",
             register: async function (text) {
-                return invoke(this.pkgname, "Register", text, "timeout while registering router")
+                return invoke(aw, pending,  this.pkgname, "Register", text, "timeout while registering router")
             },
             unregister: async function () {
-                return invoke(this.pkgname, "Unregister", "{}", "timeout while registering router")
+                return invoke(aw, pending,  this.pkgname, "Unregister", "{}", "timeout while registering router")
             }
         }
     }
 
-    return INSTANCE
-
+    // TODO wait till aries startup or send event once aries started successfully
```

Comment:
this constructor function should return a promise

Reply:
fixed and removed TODO.